src/ucp/core/ucp_mm.c

```diff
@@ -825,6 +826,7 @@ static ucs_status_t ucp_memh_init_uct_reg(ucp_context_h context, ucp_mem_h memh,
         }
 
         ucp_memh_init_from_parent(memh, cache_md_map);
+        memh->flags |= UCP_MEMH_FLAG_NON_CACHED;
```

Comment:
we set this in both if/else, should it be outside to do it only once?
or even do it from ucp_mem_map/ucp_memh_create_from_mem/ucp_memh_import when they call ucp_memh_create?